src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -199,7 +199,16 @@ private static Socket Connect(string serverName, int port, TimeSpan timeout)
             Socket[] sockets = new Socket[2];
 
             CancellationTokenSource cts = new CancellationTokenSource();
-            cts.CancelAfter(timeout);
+            
+            if (isInfiniteTimeout)
```

Comment:
Can be reduced to:

`cts.CancelAfter(isInfiniteTimeout?-1:timeout);`

Reply:
There are overloads of [`CancelAfter()`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.cancelafter?view=netframework-4.8). Using the conditional operator here won't compile : 
`Microsoft\Data\SqlClient\SNI\SNITcpHandle.cs(211,29): error CS0173: Type of conditional expression cannot be determin
ed because there is no implicit conversion between 'int' and 'System.TimeSpan'`

Reply:
Ok, yes!